File: CPP/String/Sorting/Sort Vowels in a String.cpp
class Solution {
public:
    bool isVowel(char ch) {
        if (ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U' ||
            ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u') {
            return true;
        }
        return false;
    }
    string sortVowels(string s) {
        string temp;
        for (auto c : s) {
            if (isVowel(c))
                temp += c;
        }

        sort(temp.begin(), temp.end());

        int j = 0;
        string ans;
        for (int i = 0; i < s.size(); i++) {
            if (isVowel(s[i])) {
                ans += temp[j];
                j++;
            } else {
                ans += s[i];
            }
        }
        return ans;
    }
};

File: Python/String/Sorting/Sort Vowels in a String.py
class Solution(object):
    def isVowel(self,ch):
        return ch == 'A' or ch == 'E' or ch == 'I' or ch == 'O' or ch == 'U' or ch == 'a' or ch == 'e' or ch == 'i' or ch == 'o' or ch == 'u'
    
    def sortVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        temp=[]
        for ch in s:
            if self.isVowel(ch):
                temp.append(ch)
        temp.sort()
        ans=[]
        j=0
        for i in s:
            if self.isVowel(i):
                ans.append(temp[j])
                j+=1
            else:
                ans.append(i)
        return "".join(ans)
